lib/mongo/srv/records.rb

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2014-2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'resolv'
+
+module Mongo
+
+  module SRV
+    class Records
+
+      MISMATCHED_DOMAINNAME = "Parent domain name in SRV record result (%s) does not match " +
+                                 "that of the hostname (%s)".freeze
+
+
+      attr_reader :hostname
+
+      attr_reader :hosts
+
+      attr_accessor :min_ttl
+
+      def initialize(hostname)
+        @hostname = hostname
+        @hosts = []
+        @min_ttl = nil
+      end
+
+      def empty?
+        @hosts.empty?
+      end
+
+      def add_record(record)
+        record_host = record.target.to_s
+        port = record.port
+        validate_record!(record_host, hostname)
+        @hosts << "#{record_host}#{URI::SRVProtocol::HOST_PORT_DELIM}#{port}"
+
+        if @min_ttl.nil?
+          @min_ttl = record.ttl
+        else
+          @min_ttl = [@min_ttl, record.ttl].min
+        end
+
+        self
+      end
+
+      def validate_record!(record_host, hostname)
```

Comment:
While I see spec tests for various host names, I specifically would like to also see unit tests for this method when it is given `record_host` and `host_name` with and without trailing dots. For example, something like `foo.bar.` and `bar`.

Can you please also add a docstring explaining what this method is supposed to do as it is impossible to tell from the signature.

Reply:
This method is copied verbatim from Mongo::URI::SRVProtocol; it implements the requirement from the initial seedlist discovery that the driver should error when the SRV record returns a record with a different domain name (in the fifth paragraph of [this section](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst#seedlist-discovery):

> Drivers MUST raise an error and MUST NOT initiate a connection to any returned host name which does not share the same {domainname}.